src/Nancy/NancyEngine.cs

```diff
@@ -338,5 +340,31 @@ private static void InvokeOnErrorHook(NancyContext context, ErrorPipeline pipeli
                 context.Items[ERROR_EXCEPTION] = e;
             }
         }
+
+        private static Exception FlattenException(Exception exception)
+        {
+            if (exception is AggregateException)
+            {
+                var aggregateException = exception as AggregateException;
+
+                var flattenedAggregateException = aggregateException.Flatten();
+
+                //If we have more than one exception in the AggregateException
+                //we have to send all exceptions back in order not to swallow any exceptions.
+                if (flattenedAggregateException.InnerExceptions.Count > 1)
+                {
+                    return flattenedAggregateException;
+                }
+
+                return flattenedAggregateException.InnerException;
+            }
+
+            if (exception != null && exception.InnerException != null)
+            {
+                return FlattenException(exception.InnerException);

```

Comment:
@rasmus-beck Can you remember why you're doing this? This will essentially throw away valuable information, no?